packages/agent/src/agent/http/index.ts

```diff
@@ -837,6 +839,13 @@ export class HttpAgent implements Agent {
     if (response.status === 404 && response.url.includes('api/v3')) {
       throw ProtocolError.fromCode(new HttpV3ApiNotSupportedErrorCode());
     }
+
+    if (responseText.startsWith('Invalid request expiry: ')) {
```

Comment:
I'm not entirely happy about just matching against a string like this...

Reply:
Yeah it's really not nice but I don't think we have a better way right now. We could at least include a comment that explains from where this error comes:

```suggestion
    // The error message comes from https://github.com/dfinity/ic/blob/23d5990bfc5277c32e54f0087b5a38fa412171e1/rs/validator/src/ingress_validation.rs#L233
    if (responseText.startsWith('Invalid request expiry: ')) {
```